### PR TITLE
[OPER-3362] Updates to sqltocsv

### PR DESCRIPTION
**Pull request by ehealy**
@Tapjoy/eng-group-ops 
@msc-tapjoy 

The initial purpose of this task was to add additional logging at points where gaps currently existed during the csv generation process. As the code base hadn't been modified in a number of years and the overall scope of the tool is narrow, I opted to:

* Add support for a containerized build and development process
* Update codebase to be built in go 0.12 from 0.5
* Vendor and lock dependency versions
* Add the missing logging

I have validated the code compiles in new build pipeline, a slug can be successfully built, and the newly compiled binary can be executed.

https://jenkins.tapjoy.net/job/slug-builder/56347/console

### CODE REVIEW

File: sqltocsv.go [hunk 1]
```diff
@@ -51,7 +51,10 @@ func main() {
 			if err != nil {
 				log.Fatal(err)
 			}
-			csvWriter.Write(cols)
+			write, err := csvWriter.Write(cols)
```
Comment (geoffreyclark) [inline sqltocsv.go]: When I `make build` locally I get the following error:
```
./sqltocsv.go:54:15: assignment mismatch: 2 variables but 1 values
```
My go knowledge is very limited but I suspect you need to call `write` on L58?

File: sqltocsv.go [hunk 2]
```diff
@@ -77,10 +80,16 @@ func main() {
 				log.Print(col)
 			}
 		}
-		csvWriter.Write(rowStrings)
+		write, err := csvWriter.Write(rowStrings)
```
Comment (geoffreyclark) [inline sqltocsv.go]: Same issue here:
```
./sqltocsv.go:83:14: assignment mismatch: 2 variables but 1 values
```

File: sqltocsv.go [hunk 3]
```diff
@@ -77,10 +80,16 @@ func main() {
...
+		if err != nil {
+			log.Fatal(err)
+		}
 	}
 
-	csvWriter.Flush()
+	flush, err := csvWriter.Flush()
```
Comment (geoffreyclark) [inline sqltocsv.go]: Slightly different message here but I assume it is a similar root cause:
```
./sqltocsv.go:89:31: csvWriter.Flush() used as value
```

**Review by geoffreyclark — CHANGES_REQUESTED** [review]

File: sqltocsv.go (hunk 1 above)
Comment (ehealy) [reply sqltocsv.go]: I was under the impression the write class had a built in error return, but I may have misread the documentation. Will debug further.

**Comment by ehealy** [comment]
I have updated the approach taken to error handling with the `csvWriter` object, verified the binary builds successfully and the updated code can be run without issue by performing a simple query on a util instance. 

```
2.4.1 :002 > a = `#{cmd}`
 => "" 
2.4.1 :003 > $?
 => #<Process::Status: pid 12028 exit 0> 
```

**Comment by dankleiman** [comment]
👻 👻 👻 👻 👻 👻 👻 👻 👻 👻 👻 👻 👻 👻 👻 
👻 👻 oh god how do I unsubscribe from this repo?? 👻 👻
👻 👻 👻 👻 👻 👻 👻 👻 👻 👻 👻 👻 👻 👻 👻

**Comment by ehealy** [comment]
image

**Review by geoffreyclark — APPROVED** [review]
If this is good enough for Dan, it's good enough for me.

**Pull request merged**